### PR TITLE
Simply default(T) using C# 7.1

### PR DESCRIPTION
**Pull request by smitpatel**
C# 7.1 is coming :trollface:

### CODE REVIEW

**Comment by ErikEJ** [comment]
Does this require a specific VS version?

**Comment by smitpatel** [comment]
@ErikEJ - nothing extra from what we need already.
> The C# compiler supports C# 7.1 starting with Visual Studio 2017 version 15.3, or the .NET Core SDK 2.0. However, the 7.1 features are turned off by default. To enable the 7.1 features, you need to change the language version setting for your project.

ref: https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7-1

**Comment by ErikEJ** [comment]
Ok. Then I suggest you update the "Getting and building the code" wiki page with that info.

**Comment by smitpatel** [comment]
@ErikEJ - The opt-in is per project basis. User need to specify LangVersion in csproj to get the features. Presently, all the projects in EF Core repo has C# 7.1 enabled already. So there is nothing specific is required to do for building the code. Did you mean to update something else in the wiki?

**Comment by ErikEJ** [comment]
Ah, missed that 15.3 is required already... Nothing to do then! 😄

File: src/EFCore.Relational/Query/AsyncQueryMethodProvider.cs [hunk 1]
```diff
@@ -338,18 +338,20 @@ private sealed class GroupJoinAsyncEnumerator : IAsyncEnumerator<TResult>
...
-                        _nextOuter = default(TOuter);
+                        _nextOuter = default;
                     }
 
                     if (_hasNext)
                     {
                         var outer
+#pragma warning disable IDE0034 // Simplify 'default' expression - Equals(object, object) causes default(object)
                             = Equals(_nextOuter, default(TOuter))
+#pragma warning restore IDE0034 // Simplify 'default' expression
```
Comment (bricelam) [inline src/EFCore.Relational/Query/AsyncQueryMethodProvider.cs]: Did you report this anywhere? It looks like an analyzer bug.

Comment (smitpatel) [reply src/EFCore.Relational/Query/AsyncQueryMethodProvider.cs]: https://github.com/dotnet/roslyn/issues/22505 & https://github.com/dotnet/roslyn/issues/22184 should cover it

**Pull request merged**